src/Paramore.Brighter.Extensions.DependencyInjection/ServiceCollectionBrighterBuilder.cs

```diff
@@ -233,17 +233,28 @@ from i in ti.ImplementedInterfaces
         private void RegisterAsyncMappersFromAssemblies(Assembly[] assemblies)
         {
             var mappers =
-                from ti in assemblies.SelectMany(a => a.DefinedTypes).Distinct()
+                from ti in assemblies.SelectMany(GetLoadableTypes).Distinct()
                 where ti.IsClass && !ti.IsAbstract && !ti.IsInterface
-                from i in ti.ImplementedInterfaces
+                from i in ti.GetInterfaces()
                 where i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IAmAMessageMapperAsync<>)
-                select new { RequestType = i.GenericTypeArguments.First(), HandlerType = ti.AsType() };
+                select new { RequestType = i.GenericTypeArguments.First(), HandlerType = ti };
 
             foreach (var mapper in mappers)
             {
                 _mapperRegistry.AddAsync(mapper.RequestType, mapper.HandlerType);
             }
         }
 
+        private static Type[] GetLoadableTypes(Assembly assembly)
+        {
+            try
+            {
+                return assembly.GetTypes();
+            }
+            catch (ReflectionTypeLoadException e)
+            {
+                return e.Types;
```

Comment:
This should contain the types that could be loaded if there were some types that couldn't be loaded